framework/wazuh/agent.py

```diff
@@ -676,55 +680,124 @@ def get_outdated_agents(agent_list=None, offset=0, limit=common.database_limit,
+    version : str
+        Version to upgrade to.
+    force : bool
+        force the update even if it is a downgrade.
+    use_http : bool
+        False for HTTPS protocol, True for HTTP protocol.
+    file_path : str
+        Path to the installation file.
+    installer : str
+        Selected installer.
 
-    :param agent_list: List of agents ID's.
-    :param timeout: Maximum time for the call to be considered failed.
-    :return: Upgrade result.
+    Returns
+    -------
+    ID of created tasks
     """
-    # We access unique agent_id from list, this may change if and when we decide to add option to upgrade a list of
-    # agents
-    agent_id = agent_list[0]
+    result = AffectedItemsWazuhResult(all_msg='All upgrade tasks have been created',
+                                      some_msg='Some upgrade tasks have been created',
+                                      none_msg='No upgrade task has been created',
+                                      sort_fields=['task_id'], sort_ascending='True')
+
+    agent_list = list(map(int, agents_padding(result=result, agent_list=agent_list)))
+    if not file_path:
+        wpk_repo = wpk_repo if wpk_repo else common.wpk_repo_url_4_x
+    if version and not version.startswith('v'):
+        version = f'v{version}'
+    msg = {'version': 1,
+           'origin': {'module': 'api'},
+           'command': 'upgrade' if not (installer or file_path) else 'upgrade_custom',
+           'parameters': {
+               'agents': list(),
+               'version': version,
+               'force_upgrade': force,
+               'use_http': use_http,
+               'wpk_repo': f'{wpk_repo}/' if wpk_repo and not wpk_repo.endswith('/') else wpk_repo,
+               'file_path': file_path,
+               'installer': installer
+           }
+           }
+
+    msg['parameters'] = {k: v for k, v in msg['parameters'].items() if v is not None}
+    agents_result_chunks = [agent_list[x:x + 250] for x in range(0, len(agent_list), 250)]
+
+    agent_results = list()
+    for agents_chunk in agents_result_chunks:
+        msg['parameters']['agents'] = agents_chunk
+        agent_results.append(core_upgrade_agents(command=msg))
+
+    for agent_result_chunk in agent_results:
+        for agent_result in agent_result_chunk['data']:
+            if agent_result['error'] == 0:
+                task_agent = {
+                    'agent': str(agent_result['agent']).zfill(3),
+                    'task_id': agent_result['task_id']
+                }
+                result.affected_items.append(task_agent)
+                result.total_affected_items += 1
+            else:
+                error = WazuhError(code=1810 + agent_result['error'], cmd_error=True,
+                                   extra_message=agent_result['message'])
+                result.add_failed_item(id_=str(agent_result['agent']).zfill(3), error=error)
+    result.affected_items = sorted(result.affected_items, key=lambda k: k['task_id'])
```

Comment:
We should short by agent_id by default not task_id i believe.

Reply:
Agree, sorted now by `agent`